torch/distributed/tensor/parallel/api.py

```diff
@@ -334,24 +334,25 @@ def _parallelize_multihead_attn(
         tp_multi_head_attention.copy(module)
         module = tp_multi_head_attention
 
-    if isinstance(module, TensorParallelMultiheadAttention):  # shard TPMA
-        for n, m in module.named_children():
-            if n == "qkv":
-                # Col-wise Parallelize the qkv layer.
-                distribute_module(
-                    m,
-                    device_mesh,
-                    _colwise_parallelize_linear_fn,
-                    input_fn=parallel_style._prepare_input,  # type: ignore[arg-type, misc] # pyre-ignore[6]
-                )
-            elif n == "proj":
-                # Row-wise Parallelize the proj layer
-                distribute_module(
-                    m,
-                    device_mesh,
-                    _rowwise_parallelize_linear_fn,
-                    output_fn=parallel_style._prepare_output,  # type: ignore[arg-type, misc] # pyre-ignore[6]
-                )
+    assert isinstance(module, TensorParallelMultiheadAttention)
```

Comment:
If we go for this change, let us add an assert message to provide more info?
```suggestion
    assert isinstance(module, TensorParallelMultiheadAttention), (
        f"Expects TensorParallelMultiheadAttention but got {type(module)}"
    )
```